// brunch/lib/logger.js
// Generated by CoffeeScript 1.4.0
(function() {
  'use strict';

  var color, colors, getInfo, growl, logger,
    __slice = [].slice;

  color = require('ansi-color');

  growl = require('growl');

  require('date-utils');

  colors = {
    error: 'red',
    warn: 'yellow',
    info: 'green'
  };

  getInfo = function(level) {
    var date, lvl;
    date = new Date().toFormat('DD MMM HH24:MI:SS');
    lvl = color.set(level, colors[level]);
    return "" + date + " - " + lvl + ":";
  };

  logger = {
    errorHappened: false,
    notifications: true,
    log: function() {
      var args, info, level;
      level = arguments[0], args = 2 <= arguments.length ? __slice.call(arguments, 1) : [];
      info = getInfo(level);
      return process.nextTick(function() {
        if (level === 'error' || level === 'warn') {
          return console.error.apply(console, [info].concat(__slice.call(args)));
        } else {
          return console.log.apply(console, [info].concat(__slice.call(args)));
        }
      });
    },
    error: function() {
      var args;
      args = 1 <= arguments.length ? __slice.call(arguments, 0) : [];
      if (logger.notifications) {
        growl(args.join(' '), {
          title: 'Brunch error'
        });
      }
      logger.errorHappened = true;
      return logger.log.apply(logger, ['error'].concat(__slice.call(args)));
    },
    warn: function() {
      var args;
      args = 1 <= arguments.length ? __slice.call(arguments, 0) : [];
      return logger.log.apply(logger, ['warn'].concat(__slice.call(args)));
    },
    info: function() {
      var args;
      args = 1 <= arguments.length ? __slice.call(arguments, 0) : [];
      return logger.log.apply(logger, ['info'].concat(__slice.call(args)));
    }
  };

  module.exports = Object.seal(logger);

}).call(this);
